File: ELEC881_Copilot/src/Function Name Masked/Java script (JS)/q9-partition-equal-subset-sum.js
var abc = function(nums) {
    var sum = 0;
    for (var i = 0; i < nums.length; i++) {
        sum += nums[i];
    }
    if (sum % 2 !== 0) {
        return false;
    }
    sum = sum / 2;
    var dp = new Array(sum + 1).fill(false);
    dp[0] = true;
    for (var i = 0; i < nums.length; i++) {
        for (var j = sum; j >= nums[i]; j--) {
            dp[j] = dp[j] || dp[j - nums[i]];
        }
    }
    return dp[sum];
}

File: ELEC881_Copilot/src/Function Name Masked/Python/q9-partition-equal-subset-sum.py
class abc(object):
    def abc1(self, nums):
        if sum(nums) % 2 != 0:
            return False
        target = sum(nums) // 2
        dp = [False] * (target + 1)
        dp[0] = True
        for num in nums:
            for i in range(target, num - 1, -1):
                dp[i] = dp[i] or dp[i - num]
        return dp[target]